cdmtaskservice/dockerimages.py

```diff
@@ -2,29 +2,151 @@
 Contains code for querying container information without access to a docker server
```

Comment:
It's to prevent arbitrary code execution on NERSC:

1. Create an image with tag & push to a repository
2. Register in the service after getting KBase staff blessing for the image
3. Push an entirely different image to that repository

Reply:
But I guess an `image name` + `tag` won't work?

Reply:
No, you can always push a different image to a name + tag combination. For example, you could tag & release something in GHA which would push the image, and then delete the release & tag and release and tag a completely different commit & push

Reply:
yea. I know that. But in practice, we use `image name` + `tag` most of the time. I just don't see the benefits we gain justify the ongoing maintenance required for the hash. I am fine with it if you think it's the way to go. Hopefully we are not over-building it.

Reply:
The benefit is not getting cut off by NERSC because someone was able to run malicious code on their compute because of our service

Reply:
Also if we want longer lived SFAPI tokens we'll need to pass a security review; I doubt we'll pass if they discover that someone can run any code they want once they have one version of code approved